### DISCUSSION TITLE
Subscription Message Broker

### DISCUSSION THREAD

**Discussion by Feedcube** [Netflix/dgs-framework#854] [Q&A] [answered]
Hello,
is it possible to implement the subscriptions in dgs over websockets with a message broker in spring boot?
we are having a production environment with multiple instances and for this, a message broker like rabbitmq is needed.

Thanks in advance

**Comment by srinivasankavitha** [comment, marked as answer]
Yes, that is a common use case. Once you subscribe to events on a topic, you can set up a Flux in your data fetcher and emit events based on the events coming in from the topic.
```
@DgsSubscription
public Publisher<SomeEvent> someEvent() 
return Flux.create { sink ->
            // set up a listener for the topic
            // consume and handle events from the queue
            // emit the event e.g. sink.next(it)
 }
```

Hope that answers your question.

**Reply by Feedcube** [reply]
Yes that helps!
Thank you very much
